presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -831,6 +832,66 @@ default ConnectorTableProperties getTableProperties(ConnectorSession session, Co
         return Optional.empty();
     }
 
+    /***
+     * Attempt to push down the aggregates into the table.
+     * <p>
+     * Connectors can indicate whether they don't support aggregate pushdown or that the action had no effect
+     * by returning {@link Optional#empty()}. Connectors should expect this method to be called multiple times
+     * during the optimization of a given query.
+     * <p>
+     * <b>Note</b>: it's critical for connectors to return Optional.empty() if calling this method has no effect for that
+     * invocation, even if the connector generally supports pushdown. Doing otherwise can cause the optimizer
+     * to loop indefinitely.
+     * </p>
+     * <p>
+     * If the method returns a result, the list of assignment in the result will replace the existing assignments given
+     * all the aggregations have a corresponding assignment.
+     * <p>
+     * As an example, given the following plan:
+     *
+     * <pre>
+     *
+     *  - aggregate
+     *          v0 = agg_fn1(a)
+     *          v1 = agg_fn2(b, 2)
+     *          v2 = group_by c
+     *          - scan (TH0)
+     *              a = CH0
+     *              b = CH1
+     *              c = CH2
+     * </pre>
+     * <p>
+     * The optimizer would call {@link #applyAggregation} with the following arguments:
```

Comment:
i have a warning on this line in IDE, so:
```suggestion
     * The optimizer would call this method with the following arguments:
```

Reply:
Done.